### PR TITLE
DBus Transport Bundle

### PR DESCRIPTION
**Pull request by cdjackson**
This is provided for visibility as it will require CQ.

The current DBus library is not complete implementation of the DBus protocol. It doesn't support part of the authentication handshake required to pass unix file descriptors, and then doesn't support the file descriptors when passed as args.

Currently, this is required for audio profiles, and under current BlueZ it is not possible to use TCP for this! While some changes might be looked at to provide support for TCP as well, it still seems sensible to support Unix FDs as this is a more 'out of the box' configuration.

This PR contains a number of commits to provide traceability. I started with the original file we go from Eclipse Foundation, formatted it, and then modified it. I'm happy to squash, but wanted to provide that traceability to the original file.

I'd also be surprised if this doesn't require further updates, so it's provided for visibility, discussion, and CQ kickoff (if such a thing exists)

Signed-off-by: Chris Jackson <chris@cd-jackson.com>

### CODE REVIEW

**Comment by maggu2810** [comment]
What is the academic free license? I assume we have to choose this one instead of LGPL (or at least the IP team).
I am happy to keep the commits separated but there seems something going wrong with the formatter commit (or I am missing something).
I looked at the last commit (add license and update manifest) and it contains a lot of reformat.

Without checkout and looking into it myself, what is the unix 0.5 library about?

**Comment by cdjackson** [comment]
Kai should look at this since we already have discussed this a lot and most of the license issues are already sorted or under CQ. Some of the conversation is in the other Bluetooth PR but some was private emails so probably best for Kai to look over this.

File: bundles/io/org.eclipse.smarthome.io.transport.dbus/about.html [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+        <p>
+        <strong>libdbus-java-2.7.jar</strong> <br/><br/>
+        This bundle includes a library for accessing the linux DBUS from Java. The sources can be found in the
+        <a href="https://cgit.freedesktop.org/dbus/dbus-java/">Freedesktop Git repository</a>.
+        The copyright holders have granted the Eclipse Foundation the explicit permission to distribute the
+        binary under the <a href="about_files/LICENSE">MIT license (Expat)</a>. 
+        </p>
+        <p>
+        <strong>unix-0.8.jar</strong> <br/><br/>
```
Comment (kaikreuzer) [inline bundles/io/org.eclipse.smarthome.io.transport.dbus/about.html]: the included filename is 0.5 - which version is right?

File: bundles/io/org.eclipse.smarthome.io.transport.dbus/about.html [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+and such source code may be obtained at <a href="http://www.eclipse.org/">http://www.eclipse.org</a>.</p>
+
+        <h3>Third Party Content</h3>
+        <p>The Content includes items that have been sourced from third parties as set out below. If you 
+        did not receive this Content directly from the Eclipse Foundation, the following is provided 
+        for informational purposes only, and you should look to the Redistributor's license for 
+        terms and conditions of use.</p>
+
+        <p>
+        <strong>libdbus-java-2.7.jar</strong> <br/><br/>
```
Comment (kaikreuzer) [inline bundles/io/org.eclipse.smarthome.io.transport.dbus/about.html]: If I understand this PR right, this jar is not NOT included anymore, but you have directly included the sourcecode in the bundle, is that right? That will make CQ processing more complex and I will seek assistence from my PMC colleagues on how to handle this best.

**Comment by kaikreuzer** [comment]
@maggu2810 FTR, the CQ for lib-dbus is 10561.

@cdjackson Regarding a CQ for unix, you never answered [my question here](https://github.com/eclipse/smarthome/pull/551#issuecomment-181293543) and hence I didn't yet create any CQ for it. If http://javadoc.matthew.ath.cx/projects/java/libmatthew-java-0.8.tar.gz is the right location, I can go ahead and create a CQ with these sources. 
Btw, I have just found that there are already approved CQs for https://github.com/jnr/jnr-unixsocket - but I guess this library is not usable for us, right?

**Comment by cdjackson** [comment]
Sorry - I just copied the readme over from the other repository for dbus - I'll update it.

Sorry (again) I missed your comment about the unix lib version. Well, the version I have is 0.5 - this is typically installed (or installable) directly with apt-get in unix (apt-get install libunixsocket-java) and I think if you do this, you get version 0.5.

My concern with changing to 0.8 is that if anything 'significant' has changed, then we might have an issue with distribution since we'd need to recompile the native side as well (if that interface changed). We could of course leave it out and let it be installed by the user, but that's likely to be a nightmare for installation.

Regarding the JNR version - does DT have the same concern with JNR as JNI (I guess so as there's still a native lib involved right?). If not though, this would be good since audio services in BlueZ ONLY work with Unix Sockets - TCP is not usable. Huawei have a patch, but unless it's added to BlueZ production, it will be difficult (IMHO) as we'll need to maintain separate version of BlueZ, and this is not nice - especially since BlueZ is still changing a lot for GATT! If you want, we can discuss this further offline...

Anyway, forgetting the above for a second - If it makes life easier, I could probably use this library. The DBus library probably only uses these calls in one file, so changing it would likely be easy and it's one of the files I've already changed to add support for the additional authentication handshake required.

File: bundles/io/org.eclipse.smarthome.io.transport.dbus/about.html (hunk 2 above)
Comment (cdjackson) [inline bundles/io/org.eclipse.smarthome.io.transport.dbus/about.html]: Yes - that's correct. The libdbus jar is gone, and the source is added. I used the source file that you sent me that came from Eclipse, and this is why I've kept the different commits. The first commit should pass a file compare against the original source if someone wants to verify it.

**Comment by kaikreuzer** [comment]
> Well, the version I have is 0.5

But this seems to imply that http://javadoc.matthew.ath.cx/projects/java/libmatthew-java-0.8.tar.gz is not the corresponding source code and hence I cannot use this for creating the CQ...

File: bundles/io/org.eclipse.smarthome.io.transport.dbus/about.html (hunk 2 above)
Comment (kaikreuzer) [inline bundles/io/org.eclipse.smarthome.io.transport.dbus/about.html]: My point here is that you then should not claim that a libdbus-java-2.7.jar file is embedded here.
Actually, there are probably two ways forward:
1. Maintain and modify the sources within ESH as you do with this PR. The problem is that this is not code under EPL and thus a bit tricky to deal with
2. Maintain and modify the code outside of ESH (somewhere on Github) and only include a 2.7-patched.jar in the bundle. Whenever this jar is changed, we would create a new diff-CQ for it.

**Comment by cdjackson** [comment]
Yes - that would be correct… I’ve not managed to find a definitive source for this - there are a number that I found, but none that we could say are the original authors...

File: bundles/io/org.eclipse.smarthome.io.transport.dbus/about.html (hunk 2 above)
Comment (cdjackson) [inline bundles/io/org.eclipse.smarthome.io.transport.dbus/about.html]: Agreed - I need to remove this statement. My thought was to maintain the source in this bundle...

**Comment by kaikreuzer** [comment]
Shall we try to reach out to Matthew? It seems that the IP team was at least once in contact with him already.

File: bundles/io/org.eclipse.smarthome.io.transport.dbus/about.html (hunk 2 above)
Comment (kaikreuzer) [inline bundles/io/org.eclipse.smarthome.io.transport.dbus/about.html]: I have reached out to the PMC to check whether this is a feasible option: https://dev.eclipse.org/mhonarc/lists/iot-pmc/msg01302.html

**Comment by kaikreuzer** [comment]
FTR, I created a CQ for libmatthew-java, see https://github.com/eclipse/smarthome/pull/551#issuecomment-235829224.

**Comment by kaikreuzer** [comment]
As mentioned on #551 the CQ is approved. 
@cdjackson Could you update this PR accordingly?

**Comment by kaikreuzer** [comment]
@cdjackson Did you see my last comment?

**Comment by cdjackson** [comment]
Sorry - I did see this but haven't yet looked at it. I'll look to update this over the coming few days.

**Comment by kaikreuzer** [comment]
@cdjackson Ping :-)

**Comment by bokas74** [comment]
Hi any updates to the DBus libraries? I tried testing it on a RPi3 (after updating to Bluez 5.41), but didn't seem to work. Trying to debug.

Reg's
Anirban

**Comment by cdjackson** [comment]
The DBus libraries won’t change function - the current issue is simply to move them from the source to attached JAR, so I doubt it will change your problem.

**Comment by cdjackson** [comment]
Source removed + JAR added.

File: bundles/io/org.eclipse.smarthome.io.transport.dbus/build.properties [hunk 3]
```diff
@@ -0,0 +1,8 @@
+bin.includes = META-INF/,\
+               .,\
+               about.html,\
+               lib/unix-0.5.jar,\
```
Comment (kaikreuzer) [inline bundles/io/org.eclipse.smarthome.io.transport.dbus/build.properties]: looks as if you are missing the second jar here?

**Comment by cdjackson** [comment]
Updated.

**Comment by kaikreuzer** [comment]
Thanks! So next up is the BLE binding :-)

**Pull request merged**